tests/by-util/test_env.rs

```diff
@@ -882,6 +882,33 @@ fn test_env_arg_ignore_signal_empty() {
         .no_stderr()
         .stdout_contains("hello");
 }
+
+#[test]
+fn disallow_equals_sign_on_short_options() {
+    let ts = TestScenario::new(util_name!());
+
+    ts.ucmd()
+        .arg("-u=")
+        .fails()
+        .code_is(125)
+        .stderr_contains("env: cannot unset '=': Invalid argument");
+    ts.ucmd()
+        .arg("-u=A1B2C3")
+        .fails()
+        .code_is(125)
+        .stderr_contains("env: cannot unset '=A1B2C3': Invalid argument");
+    ts.ucmd().arg("--split-string=A1B=2C3=").succeeds();
+    ts.ucmd()
+        .arg("--unset=")
+        .fails()
+        .code_is(125)
+        .stderr_contains("env: cannot unset '': Invalid argument");
+    ts.ucmd().arg("A1B=2C3").succeeds();
+    ts.ucmd().arg("--unset=A1B2C3").succeeds();
+    ts.ucmd().arg("--unset=A1B23C").succeeds();
+    ts.ucmd().arg("-u A1B23C").succeeds();
```

Comment:
Here I don't understand what you are trying to test.

Reply:
Those tests were to make sure even though -u= is not allowed it does not mess with --unset at all or -u without '='. The second --unset is redundant, I think I accidentally duplicated that one. Would it be better if I got rid of these specific tests since they're not directly affected by my change?

Reply:
That makes sense, thank you! I've committed a change to remove them.